### PR TITLE
Implement IS_NULL operator and INTEGER value type

### PR DESCRIPTION
**Pull request by typotter**
- `IS_NULL` operator type and matching condition
- `INTEGER` value type and corresponding client method
- small refactorings.

### CODE REVIEW

File: dot-net-sdk/EppoClient.cs [hunk 1]
```diff
@@ -59,7 +65,7 @@ public class EppoClient
         }
 
         var subjectVariationOverride = this.GetSubjectVariationOverride(subjectKey, configuration);
-        if (!subjectVariationOverride.isNull())
+        if (!subjectVariationOverride.IsNull())
```
Comment (leoromanovsky) [inline dot-net-sdk/EppoClient.cs:68]: 👍

File: dot-net-sdk/EppoClient.cs [hunk 2]
```diff
@@ -40,6 +40,12 @@ public class EppoClient
     }
 
 
+    public double? GetIntegerAssignment(string subjectKey, string flagKey, SubjectAttributes? subjectAttributes = null)
```
Comment (leoromanovsky) [inline dot-net-sdk/EppoClient.cs:43]: You got it returning a `double` did you mean an `int` or `long` type ?

File: dot-net-sdk/dto/EppoValueDeserializer.cs [hunk 3]
```diff
@@ -18,14 +18,15 @@ public override void WriteJson(JsonWriter writer, EppoValue? value, JsonSerializ
...
-                return new EppoValue(value.ToString(), EppoValueType.STRING);
+                return EppoValue.String(value.ToString());
             case JsonToken.Integer:
+                return EppoValue.Integer(value.ToString());
             case JsonToken.Float:
-                return new EppoValue(value.ToString(), EppoValueType.NUMBER);
+                return EppoValue.Number(value.ToString());
             case JsonToken.Boolean:
-                return new EppoValue(value.ToString(), EppoValueType.BOOLEAN);
+                return EppoValue.Bool(value.ToString());
```
Comment (leoromanovsky) [inline dot-net-sdk/dto/EppoValueDeserializer.cs:27]: 👍

File: dot-net-sdk/http/ExperimentConfigurationRequester.cs [hunk 4]
```diff
@@ -15,20 +15,15 @@ public class ExperimentConfigurationRequester
...
-        {
-            throw e;
-        }
         catch (Exception e)
         {
             logger.Warn($"Unable to Fetch Experiment Configuration: {e.Message}");
         }
 
-        return config;
+        return null;
```
Comment (leoromanovsky) [inline dot-net-sdk/http/ExperimentConfigurationRequester.cs:27]: Thanks I like this style more as well - mutating variables is hard to follow.

**Review by schmit — APPROVED** [review]
Little experience with .net but changes make sense to me!

File: dot-net-sdk/validators/RuleValidator.cs [hunk 5]
```diff
@@ -28,14 +28,18 @@ private static bool EvaluateCondition(SubjectAttributes subjectAttributes, Condi
     {
         try
         {
-            if (subjectAttributes.ContainsKey(condition.attribute) &&
-                subjectAttributes.TryGetValue(condition.attribute, out EppoValue outVal))
+            // Operators other than `IS_NULL` need to assume non-null
+            if (condition.operatorType == IS_NULL) {
+                bool isNull = !subjectAttributes.TryGetValue(condition.attribute, out EppoValue outVal) || outVal.IsNull() || outVal.value == null;
```
Comment (leoromanovsky) [inline dot-net-sdk/validators/RuleValidator.cs:33]: check if `outVal` is null here before access in `outVal.value`

File: dot-net-sdk/EppoClient.cs (hunk 2 above)
Comment (typotter) [reply dot-net-sdk/EppoClient.cs:43]: ya, sure did. thanks for catching

**Comment by typotter** [comment]
Thanks for the review. Updates made. @leoromanovsky PTAL

**Review by leoromanovsky — APPROVED** [review]

**Comment by typotter** [comment]
Thanks @leoromanovsky and @schmit
